Add RobustMultiHelix pat. rec.

brownd1978 opened the pull request:
This  PR replaces PR 25 with a more modular configuration structure.  I suggest making this a standard pattern in future.

gianipez commented:
Hello @brownd1978 , I think we should stick to the current organization for the fcl files. The change you proposed reflects a design that was discussed initially and we opted for a different one. 
I'm reposting the various other comments from the previous PR.  

As I already said, I would like to check the trigger counts when you run the new menu over the full NoPrimary1BB sample before merging this PR using a recent version of the Offline.

brownd1978 commented:
Hi Gianni, this needs further discussion.   Putting all the trigger
configurations in one file makes them unmaintainable and unreadable.   What
possible advantage is there to this choice.

David Brown            ***@***.***
Office Phone (510) 486-7261
Lawrence Berkeley National Lab
M/S 50R5008 (50-6026C)  Berkeley, CA 94720


On Fri, May 24, 2024 at 04:38 Gianantonio Pezzullo ***@***.***>
wrote:

> Hello @brownd1978 <https://github.com/brownd1978> , I think we should
> stick to the current organization for the fcl files. The change you
> proposed reflects a design that was discussed initially and we opted for a
> different one.
> I'm reposting the various other comments from the previous PR.
>
> As I already said, I would like to check the trigger counts when you run
> the new menu over the full NoPrimary1BB sample before merging this PR using
> a recent version of the Offline.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/mu2e_trig_config/pull/27#issuecomment-2129314474>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH577ZL7B6UQJEHBFDHGTZD4RCBAVCNFSM6AAAAABIG6UKS6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCMRZGMYTINBXGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

gianipez commented:
I 

> Hi Gianni, this needs further discussion. Putting all the trigger configurations in one file makes them unmaintainable and unreadable. What possible advantage is there to this choice. David Brown ***@***.*** Office Phone (510) 486-7261 Lawrence Berkeley National Lab M/S 50R5008 (50-6026C) Berkeley, CA 94720
> […](#)

PRs are not ideal for proposing a restructuring of the repository together with trivial changes. I see issues with your proposed re-organization that I don't think we should discuss here.

brownd1978 commented:
I integrated the mpr content back into the monolithic files as requested.  I don't understand the following comment:
> I would like to check the trigger counts when you run the new menu over the full NoPrimary1BB sample before merging this PR using a recent version of the Offline.
Are you asking me to do something here?
As for the lowp, etc triggers: they are not essential for the current tasks and can be added later. There is also a hack in these scripts to accomodate the fact that RobustMultiHelixFinder has not been migrated to produce a merged DeM + DeP output collection.  Once that is fixed (in Offline) I can remove the hack and look into other triggers.  But that can be a separate PR.

gianipez commented on core/trigFilters.fcl:
```diff
@@ -1280,10 +1280,105 @@ TrigAprFilters:{
+    helixSeedCollection: "TTMprHelixMergerDe"
+    posHelicitySelection : {
+      configured: true
+      doHelicityCheck: false
```
this should be set to true because you define both helicity selections

brownd1978 replied:
OK

gianipez commented on core/trigProducers.fcl:
```diff
@@ -663,59 +659,122 @@ TrigCprProducers : {
+    DiagLevel               : 0
+  }
+  TTMprHelixMergerDe: {
+    HelixFinders: [ "TTRobustMultiHelixFinder:Positive" ]
```
this should include both helicities in the list

brownd1978 replied:
good point.  Fixed.

gianipez reviewed (commented):
I made a few comments that should be easy to implement. 
Once you made the requested changes, can you also run the triggerTest job over the NoPrimary 1BB dataset and post the trigger counts summary here?

brownd1978 commented on data/physMenu.json:
```diff
@@ -156,15 +165,15 @@
             "enabled": 1,
-            "eventModeConfig" : [{"eventMode":"OnSpill", "prescale":10000, "streams": [ "physics_minBias" ]},
-                                 {"eventMode":"OffSpill", "prescale":10000, "streams": [ "cosmics_main", "physics_minBias" ]}]
+            "eventModeConfig" : [{"eventMode":"OnSpill", "prescale":1e4, "streams": [ "physics_minBias" ]},
```
that was an error introduced by the merge, now fixed.

brownd1978 commented on core/trigMpr.fcl:
```diff
@@ -0,0 +1,163 @@
+      helixSeedCollection: "TTMprHelixMergerDe"
+      posHelicitySelection : {
+        configured: true
+        doHelicityCheck: false
```
OK.

brownd1978 commented on core/trigMpr.fcl:
```diff
@@ -0,0 +1,163 @@
+  #  Producers
+  #
+  Producers : {
+    TTRobustMultiHelixFinder : {
```
Yes, that is on the issue list with the Mpr author.  It will happen in a subsequent (Offline) PR.

brownd1978 commented on data/physMenu.json:
```diff
@@ -8,13 +8,22 @@
 
     "trigger_paths" : {
 
+        "mprDe_highP_stopTarg": {
```
I can add it if you prefer.

gianipez replied:
thanks

brownd1978 commented:
I believe I've addressed all the comments.  I ran triggerTest.fcl on NoPrimary as requested, the log file is in /nashome/b/brownd/mu2e/KFTrack/ttnoprimary.log
